oauth2client/contrib/gce.py

```diff
@@ -85,32 +138,128 @@ class AppAssertionCredentials(AssertionCredentials):
     information to generate and refresh its own access tokens.
     """
 
-    @util.positional(2)
-    def __init__(self, scope='', **kwargs):
+    def __init__(self,
+                 scope=None,
+                 service_account_email='default',
+                 service_account_info=None,
+                 **unused_kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
-            scope: string or iterable of strings, scope(s) of the credentials
-                   being requested. Using this argument will have no effect on
-                   the actual scopes for tokens requested. These scopes are
-                   set at VM instance creation time and won't change.
+            scope:
+                string or iterable of strings, scope(s) of the credentials
+                being requested. Using this argument will have no effect on
+                the actual scopes for tokens requested. These scopes are
+                set at VM instance creation time and won't change.
+            service_account_email:
+                the email for these credentials. This can be used with custom
+                service accounts, or left blank to use the default service
+                account for the instance. The default service account is
+                usually the shared Compute Engine service account.
+            service_account_info:
+                (Optional) Dictionary containing information about a
+                service account (typically deserialized from JSON).
+                If passed in, will be used as ``service_account_info``
+                property. Otherwise, the Compute Engine metadata server
+                will be used to determine this value. Additionally,
+                if present the 'email' field in this argument must be
+                provided and will override the ``service_account_email``
+                argument.
         """
-        if scope:
-            warnings.warn(_SCOPES_WARNING)
-        # This is just provided for backwards compatibility, but is not
-        # used by this class.
-        self.scope = util.scopes_to_string(scope)
-        self.kwargs = kwargs
+
+        self._service_account_info = service_account_info or {
```

Comment:
The argument `service_account_info` is only provided for serialization and deserialization, and not intended for end user use. I could remove it from the constructor and do,

```
def from_json(cls, json_data):
    creds = cls()
    creds._service_account_info = json_data['service_account_info']
```
If you prefer